In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [ ]:
#GitHub url
url = ("https://raw.githubusercontent.com/senumikumarawadu-ctrl/SDI_Assignment/refs/heads/main/The_Migration_Mindset_of_IT_Graduates_in_Sri_Lanka.csv")

#Read the Excel file
df= pd.read_csv(url)

#Display the first 5 rows
print("First 5 Rows of the Dataset")
display(df.head())

#Display dataset information
print("/nDataset Information")
df.info()



First 5 Rows of the Dataset


,age,gender,marital_status,breadwinner,num_of_dep,uni_type,academic_spl,final_gpa,job_role,workplace,...,monthly_salary,percentage_of_living_expenses,current_job_role_sat_rate,family_living_abroad,relatives_abroad,friends_it_migrated,plan_to_migrate,primary_reasons_for_migrating,timeframe_planning_to_migrate,preferred_country
0,32,Male,Unmarried,No,0,Government (State),Computer Science,3.27,Entrepreneur,Beep technologies pvt ltd,...,"Below 200,000",40% - 60%,1,No,0,20,Undecided,Not applicable,Not applicable,Not applicable
1,28,Male,Unmarried,No,4,Government (State),Electronics and Information technology,3.40,QA Engineer / UI/UX Designer,MillenniumIT,...,"200,000 - 500,000",40% - 60%,3,No,3,10,Undecided,Not applicable,Not applicable,Not applicable
2,24,Male,Unmarried,No,2,Private,Computer Science,3.00,Software Engineer / Developer,LOLC Technologies,...,"Below 200,000",My expenditures exceed my income,2,No,0,5,Yes,Higher Salaries and Better Financial Prospects...,Needs review,Schengen Countries
3,30,Female,Married,No,4,Government (State),Computer Science and Industrial Management,3.24,Project Manager,Infor Sri Lanka,...,"200,000 - 500,000",40% - 60%,5,No,10,8,Undecided,Not applicable,Not applicable,Not applicable
4,29,Male,Married,No,0,Private,Software Engineering,3.69,Software Engineer / Developer,IFS,...,"200,000 - 500,000",61% - 80%,3,No,15,10,Undecided,Not applicable,Not applicable,Not applicable


/nDataset Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96 entries, 0 to 95
Data columns (total 22 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   age                            96 non-null     int64  
 1   gender                         96 non-null     object 
 2   marital_status                 96 non-null     object 
 3   breadwinner                    96 non-null     object 
 4   num_of_dep                     96 non-null     int64  
 5   uni_type                       96 non-null     object 
 6   academic_spl                   96 non-null     object 
 7   final_gpa                      95 non-null     float64
 8   job_role                       96 non-null     object 
 9   workplace                      96 non-null     object 
 10  experience                     96 non-null     float64
 11  num_certs                      96 non-null     int64  
 12  monthly_salary                

In [ ]:
#Numerical variable 1 - Age
print("Summary Statistics for Age:")
col = df["age"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Age:
n=96, mean=30.00, median=28.00, mode=27
min=19, max=54, range=35
variance=50.67, std_dev=7.12
skewness=1.40, kurtosis=1.90


In [ ]:
age = df["age"]

print("\nSummary Statistics for Age:")
display(age.describe())


Summary Statistics for Age:


,age
count,96.000000
mean,30.000000
std,7.118545
min,19.000000
25%,25.000000
50%,28.000000
75%,32.000000
max,54.000000


In [ ]:
#Numerical variable 2 - Number of Dependents
print("Summary Statistics for Number of Dependents:")
col = df["num_of_dep"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Number of Dependents:
n=96, mean=1.35, median=1.00, mode=0
min=0, max=6, range=6
variance=2.53, std_dev=1.59
skewness=0.88, kurtosis=-0.29


In [ ]:
num_of_dep = df["num_of_dep"]

print("\nSummary Statistics for Number of Dependents:")
display(num_of_dep.describe())


Summary Statistics for Number of Dependents:


,num_of_dep
count,96.000000
mean,1.354167
std,1.589301
min,0.000000
25%,0.000000
50%,1.000000
75%,3.000000
max,6.000000


In [ ]:
#Numerical variable 3 - GPA
print("Summary Statistics for GPA:")
col = df["final_gpa"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for GPA:
n=95, mean=3.33, median=3.40, mode=3.2
min=1.9, max=4.0, range=2.1
variance=0.19, std_dev=0.44
skewness=-1.09, kurtosis=1.58


In [ ]:
final_gpa = df["final_gpa"]

print("\nSummary Statistics for GPA:")
display(final_gpa.describe())


Summary Statistics for GPA:


,final_gpa
count,95.000000
mean,3.325368
std,0.437434
min,1.900000
25%,3.100000
50%,3.400000
75%,3.690000
max,4.000000


In [ ]:
#Numerical variable 4 - Experience
print("Summary Statistics for Experience in IT industry:")
col = df["experience"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Experience in IT industry:
n=96, mean=5.95, median=4.00, mode=1.0
min=0.5, max=32.0, range=31.5
variance=42.71, std_dev=6.54
skewness=2.09, kurtosis=4.28


In [ ]:
experience = df["experience"]

print("\nSummary Statistics for experience:")
display(experience.describe())


Summary Statistics for experience:


,experience
count,96.000000
mean,5.947917
std,6.535119
min,0.500000
25%,1.875000
50%,4.000000
75%,7.000000
max,32.000000


In [ ]:
#Numerical variable 5 - Number of Certifications completed
print("Summary Statistics for Number of Certifications completed:")
col = df["num_certs"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Number of Certifications completed:
n=96, mean=4.84, median=3.00, mode=2
min=0, max=27, range=27
variance=25.35, std_dev=5.04
skewness=2.01, kurtosis=4.42


In [ ]:
num_certs = df["num_certs"]

print("\nSummary Statistics for Number of Certifications Completed:")
display(num_certs.describe())


Summary Statistics for Number of Certifications Completed:


,num_certs
count,96.000000
mean,4.843750
std,5.035303
min,0.000000
25%,2.000000
50%,3.000000
75%,6.000000
max,27.000000


In [ ]:
#Numerical variable 6 - Number of Relatives Living Abroad
print("Summary Statistics for Number of Relatives Living Abroad:")
col = df["relatives_abroad"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Number of Relatives Living Abroad:
n=96, mean=3.89, median=3.00, mode=0
min=0, max=16, range=16
variance=15.07, std_dev=3.88
skewness=1.18, kurtosis=0.94


In [ ]:
relatives_abroad = df["relatives_abroad"]

print("\nSummary Statistics for Number of Relatives Living abroad:")
display(relatives_abroad.describe())


Summary Statistics for Number of Relatives Living abroad:


,relatives_abroad
count,96.000000
mean,3.885417
std,3.882131
min,0.000000
25%,1.000000
50%,3.000000
75%,6.000000
max,16.000000


In [ ]:
#Numerical variable 7 - Number of close friends in IT job role who have already migrated
print("Summary Statistics for Number of close friends in IT job role who have already migrated:")
col = df["friends_it_migrated"]

n = col.count()
mean = col.mean()
median = col.median()
mode = col.mode()[0]
minimum = col.min()
maximum = col.max()
range_ = maximum - minimum
variance = col.var()
std_dev = col.std()
skewness = col.skew()
kurtosis = col.kurt()

print(f"n={n}, mean={mean:.2f}, median={median:.2f}, mode={mode}")
print(f"min={minimum}, max={maximum}, range={range_}")
print(f"variance={variance:.2f}, std_dev={std_dev:.2f}")
print(f"skewness={skewness:.2f}, kurtosis={kurtosis:.2f}")

Summary Statistics for Number of close friends in IT job role who have already migrated:
n=96, mean=7.78, median=6.00, mode=10
min=0, max=100, range=100
variance=116.83, std_dev=10.81
skewness=6.75, kurtosis=56.62


In [ ]:
friends_it_migrated = df["friends_it_migrated"]

print("\nSummary Statistics for Number of close friends in IT job role who have already migrated:")
display(friends_it_migrated.describe())


Summary Statistics for Number of close friends in IT job role who have already migrated:


,friends_it_migrated
count,96.000000
mean,7.781250
std,10.808577
min,0.000000
25%,3.000000
50%,6.000000
75%,10.000000
max,100.000000


In [ ]:
from scipy import stats
from scipy.stats import t, norm

1.Is the average final GPA of the participants greater than
 3.3
?


In [ ]:
# Final GPA
final_gpa = df["final_gpa"].dropna()

# One-sample t-test against 3.3
t_stat, p_two = stats.ttest_1samp(final_gpa, popmean=3.3)

alpha = 0.05
df_gpa = len(final_gpa) - 1

# Convert to one-tailed p-value and calculate critical t-value for right-tailed test
if t_stat > 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(1 - alpha, df_gpa)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(alpha, df_gpa) # Though this case is less likely for 'greater than' hypothesis

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(final_gpa),
        final_gpa.mean(),
        final_gpa.std(),
        3.3,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,95.0000
1,Sample Mean,3.3254
2,Sample Standard Deviation,0.4374
3,Hypothesized Mean,3.3000
4,Critical t-value,1.6612
5,t Statistic,0.5653
6,One-tailed p-value,0.2866


Here is the evaluation for the first question.

**Q1. Is the average final GPA of the participants greater than 3.3?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\le3.3$
(The average final GPA of the population is less than or equal to 3.3.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu>3.3$
(The average final GPA of the population is greater than 3.3.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the provided summary values:

* Sample Size ($n$) = 95
* Sample Mean ($\bar{x}$) = 3.3254
* Sample Standard Deviation ($s$) = 0.4374
* Hypothesized Mean ($\mu_0$) = 3.3000

$t=\frac{3.3254-3.3000}{0.4374/\sqrt{95}} \approx 0.5653$
Test Statistic ($t$): 0.5653

5. Critical Value
For a right-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): 1.6612
6. P-Value
One-tailed P-value: 0.2866
7. Decision
Fail to reject $H_0$ (Do not reject the null hypothesis).

* P-value rule: $P\text{-value}=0.2866 > \alpha=0.05$.
* Critical value rule: $t=0.5653 < t_{critical}=1.6612$ (the test statistic does not fall into the critical rejection region).

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.2866) is greater than the significance level ($\alpha=0.05$), we fail to reject $H_0$. This means that the probability of obtaining a sample mean of 3.3254 or higher purely by random chance when the true population mean is 3.3 is roughly 28.66%. Since this chance is well above our 5% threshold, there is no statistically significant evidence to support $H_a$.
Verdict: No, we cannot conclude that the average final GPA of the participants is greater than 3.3.



2.Is the average age of respondents greater than 28 years?


In [ ]:
# Age
age = df["age"].dropna()

# One-sample t-test against 28
t_stat, p_two = stats.ttest_1samp(age, popmean=28)

alpha = 0.05
df_age = len(age) - 1

# Convert to one-tailed p-value and calculate critical t-value for right-tailed test
if t_stat > 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(1 - alpha, df_age)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(alpha, df_age) # Though this case is less likely for 'greater than' hypothesis

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(age),
        age.mean(),
        age.std(),
        28,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,30.0000
2,Sample Standard Deviation,7.1185
3,Hypothesized Mean,28.0000
4,Critical t-value,1.6611
5,t Statistic,2.7528
6,One-tailed p-value,0.0035


**Q2. Is the average age of respondents greater than 28 years?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\le28$
(The average age of the population is less than or equal to 28.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu>28$
(The average age of the population is greater than 28.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the provided summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 30.0000
* Sample Standard Deviation ($s$) = 7.1185
* Hypothesized Mean ($\mu_0$) = 28.0000

$t=\frac{30.0000-28.0000}{7.1185/\sqrt{96}} \approx 2.7528$
Test Statistic ($t$): 2.7528

5. Critical Value
For a right-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): 1.6611
6. P-Value
One-tailed P-value: 0.0035
7. Decision
Reject $H_0$.

* P-value rule: $P\text{-value}=0.0035 < \alpha=0.05$.
* Critical value rule: $t=2.7528 > t_{critical}=1.6611$.

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.0035) is less than the significance level ($\alpha=0.05$), we reject $H_0$.
Verdict: Yes, we can conclude that the average age of the respondents is greater than 28 years.



3.Does average number of certifications completed differ from 3?


In [ ]:
# Number of Certifications
num_certs = df["num_certs"].dropna()

# One-sample t-test against 3
t_stat, p_two = stats.ttest_1samp(num_certs, popmean=3)

alpha = 0.05
df_num_certs = len(num_certs) - 1

# Calculate critical t-values for a two-tailed test
lower_critical_t = t.ppf(alpha / 2, df_num_certs)
upper_critical_t = t.ppf(1 - (alpha / 2), df_num_certs)

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Lower Critical t-value",
        "Upper Critical t-value",
        "t Statistic",
        "two-tailed p-value"
    ],
    "Value":[
        len(num_certs),
        num_certs.mean(),
        num_certs.std(),
        3,
        lower_critical_t,
        upper_critical_t,
        t_stat,
        p_two
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,4.8438
2,Sample Standard Deviation,5.0353
3,Hypothesized Mean,3.0000
4,Lower Critical t-value,-1.9853
5,Upper Critical t-value,1.9853
6,t Statistic,3.5877
7,two-tailed p-value,0.0005


**Q3. Does average number of certifications completed differ from 3?**

1. Null Hypothesis ($H_0$)
$H_0:\mu=3$
(The average number of certifications completed is equal to 3.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu\neq3$
(The average number of certifications completed differs from 3.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the provided summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 4.8438
* Sample Standard Deviation ($s$) = 5.0353
* Hypothesized Mean ($\mu_0$) = 3.0000

$t=\frac{4.8438-3.0000}{5.0353/\sqrt{96}} \approx 3.5877$
Test Statistic ($t$): 3.5877

5. Critical Value
For a two-tailed t-test at $\alpha=0.05$:
Lower Critical Value ($t_{critical}$): -1.9853
Upper Critical Value ($t_{critical}$): 1.9853
6. P-Value
Two-tailed P-value: 0.0005
7. Decision
Reject $H_0$.

* P-value rule: $P\text{-value}=0.0005 < \alpha=0.05$.
* Critical value rule: $t=3.5877 > t_{critical}=1.9853$.

8. Conclusion
Connection of P-Value against $H_0$: Because the two-tailed P-value (0.0005) is less than the significance level ($\alpha=0.05$), we reject $H_0$.
Verdict: Yes, we can conclude that the average number of certifications completed differs significantly from 3.



4.Is the average years of experience greater than 5 years?

In [ ]:
# Years of Experience
experience = df["experience"].dropna()

# One-sample t-test against 5
t_stat, p_two = stats.ttest_1samp(experience, popmean=5)

alpha = 0.05
df_experience = len(experience) - 1

# Convert to one-tailed p-value and calculate critical t-value for right-tailed test
if t_stat > 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(1 - alpha, df_experience)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(alpha, df_experience) # Though this case is less likely for 'greater than' hypothesis

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(experience),
        experience.mean(),
        experience.std(),
        5,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,5.9479
2,Sample Standard Deviation,6.5351
3,Hypothesized Mean,5.0000
4,Critical t-value,1.6611
5,t Statistic,1.4212
6,One-tailed p-value,0.0793


**Q4. Is the average years of experience greater than 5 years?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\le5$
(The average years of experience is less than or equal to 5.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu>5$
(The average years of experience is greater than 5.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the provided summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 5.9479
* Sample Standard Deviation ($s$) = 6.5351
* Hypothesized Mean ($\mu_0$) = 5.0000

$t=\frac{5.9479-5.0000}{6.5351/\sqrt{96}} \approx 1.4212$
Test Statistic ($t$): 1.4212

5. Critical Value
For a right-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): 1.6611
6. P-Value
One-tailed P-value: 0.0793
7. Decision
Fail to reject $H_0$.

* P-value rule: $P\text{-value}=0.0793 > \alpha=0.05$.
* Critical value rule: $t=1.4212 < t_{critical}=1.6611$.

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.0793) is greater than $\alpha=0.05$, we fail to reject $H_0$. This means that there is a 7.93% probability of obtaining a sample mean of 5.9479 or higher purely by random chance when the true population mean is 5. Since this chance exceeds our 5% threshold, there is not enough evidence to support $H_a$.
Verdict: No, we cannot conclude that the average years of experience is greater than 5 years.



5.Is the average number of relatives living abroad less than 5?

In [ ]:
# Number of Relatives Living Abroad
relatives_abroad = df["relatives_abroad"].dropna()

# One-sample t-test against 5
t_stat, p_two = stats.ttest_1samp(relatives_abroad, popmean=5)

alpha = 0.05
df_relatives = len(relatives_abroad) - 1

# Convert to one-tailed p-value and calculate critical t-value for left-tailed test
if t_stat < 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(alpha, df_relatives)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(1 - alpha, df_relatives) # Though this case is less likely for 'less than' hypothesis

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(relatives_abroad),
        relatives_abroad.mean(),
        relatives_abroad.std(),
        5,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,3.8854
2,Sample Standard Deviation,3.8821
3,Hypothesized Mean,5.0000
4,Critical t-value,-1.6611
5,t Statistic,-2.8131
6,One-tailed p-value,0.0030


**Q5. Is the average number of relatives living abroad less than 5?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\ge5$
(The average number of relatives living abroad is greater than or equal to 5.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu<5$
(The average number of relatives living abroad is less than 5.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the provided summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 3.8854
* Sample Standard Deviation ($s$) = 3.8821
* Hypothesized Mean ($\mu_0$) = 5.0000

$t=\frac{3.8854-5.0000}{3.8821/\sqrt{96}} \approx -2.8131$
Test Statistic ($t$): -2.8131

5. Critical Value
For a left-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): -1.6611
6. P-Value
One-tailed P-value: 0.0030
7. Decision
Reject $H_0$.

* P-value rule: $P\text{-value}=0.0030 < \alpha=0.05$.
* Critical value rule: $t=-2.8131 < t_{critical}=-1.6611$.

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.0030) is less than the significance level ($\alpha=0.05$), we reject $H_0$. This means that the probability of obtaining a sample mean of 3.8854 or lower purely by random chance when the true population mean is 5 is roughly 0.3%. Since this chance is well below our 5% threshold, there is statistically significant evidence to support $H_a$.
Verdict: Yes, we can conclude that the average number of relatives living abroad is significantly less than 5.



6.Is the average number of close friends in IT who have migrated greater than 5?

In [ ]:
# Number of Close Friends in IT who Migrated
friends_it_migrated = df["friends_it_migrated"].dropna()

# One-sample t-test against 5
t_stat, p_two = stats.ttest_1samp(friends_it_migrated, popmean=5)

alpha = 0.05
df_friends = len(friends_it_migrated) - 1

# Convert to one-tailed p-value and calculate critical t-value for right-tailed test
if t_stat > 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(1 - alpha, df_friends)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(alpha, df_friends) # Though this case is less likely for 'greater than' hypothesis

# Sample statistics
results = pd.DataFrame({
   "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(friends_it_migrated),
        friends_it_migrated.mean(),
        friends_it_migrated.std(),
        5,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,7.7812
2,Sample Standard Deviation,10.8086
3,Hypothesized Mean,5.0000
4,Critical t-value,1.6611
5,t Statistic,2.5212
6,One-tailed p-value,0.0067


**Q6. Is the average number of close friends in IT who have migrated greater than 5?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\le5$
(The average number of close friends in IT who migrated is less than or equal to 5.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu>5$
(The average number of close friends in IT who migrated is greater than 5.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 7.7812
* Sample Standard Deviation ($s$) = 10.8086
* Hypothesized Mean ($\mu_0$) = 5.0000

$t=\frac{7.7812-5.0000}{10.8086/\sqrt{96}} \approx 2.5212$
Test Statistic ($t$): 2.5212

5. Critical Value
For a right-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): 1.6611
6. P-Value
One-tailed P-value: 0.0067
7. Decision
Reject $H_0$.

* P-value rule: $P\text{-value}=0.0067 < \alpha=0.05$.
* Critical value rule: $t=2.5212 > t_{critical}=1.6611$.

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.0067) is less than the significance level ($\alpha=0.05$), we reject $H_0$. This means that the probability of obtaining a sample mean of 7.7812 or higher purely by random chance when the true population mean is 5 is roughly 0.67%. Since this chance is well below our 5% threshold, there is statistically significant evidence to support $H_a$.
Verdict: Yes, we can conclude that the average number of close friends in IT who have migrated is greater than 5.



7.Is the average number of dependents less than 2?

In [ ]:
# Number of Dependents
num_of_dep = df["num_of_dep"].dropna()

# One-sample t-test against 2
t_stat, p_two = stats.ttest_1samp(num_of_dep, popmean=2)

alpha = 0.05
df_dependents = len(num_of_dep) - 1

# Convert to one-tailed p-value and calculate critical t-value for left-tailed test
if t_stat < 0:
    p_one = p_two / 2
    critical_t_value = t.ppf(alpha, df_dependents)
else:
    p_one = 1 - (p_two / 2)
    critical_t_value = t.ppf(1 - alpha, df_dependents) # Though this case is less likely for 'less than' hypothesis

# Sample statistics
results = pd.DataFrame({
    "Statistic":[
        "Sample Size",
        "Sample Mean",
        "Sample Standard Deviation",
        "Hypothesized Mean",
        "Critical t-value",
        "t Statistic",
        "One-tailed p-value"
    ],
    "Value":[
        len(num_of_dep),
        num_of_dep.mean(),
        num_of_dep.std(),
        2,
        critical_t_value,
        t_stat,
        p_one
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,1.3542
2,Sample Standard Deviation,1.5893
3,Hypothesized Mean,2.0000
4,Critical t-value,-1.6611
5,t Statistic,-3.9815
6,One-tailed p-value,0.0001


**Q7. Is the average number of dependents less than 2?**

1. Null Hypothesis ($H_0$)
$H_0:\mu\ge2$
(The average number of dependents is greater than or equal to 2.)
2. Alternative Hypothesis ($H_a$)
$H_a:\mu<2$
(The average number of dependents is less than 2.)
3. Significance Level ($\alpha$)
$\alpha=0.05$
4. Test Statistic
Using the formula for the one-sample t-statistic with sample standard deviation ($s$):
$t=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}$
Substituting the summary values:

* Sample Size ($n$) = 96
* Sample Mean ($\bar{x}$) = 1.3542
* Sample Standard Deviation ($s$) = 1.5893
* Hypothesized Mean ($\mu_0$) = 2.0000

$t=\frac{1.3542-2.0000}{1.5893/\sqrt{96}} \approx -3.9815$
Test Statistic ($t$): -3.9815

5. Critical Value
For a left-tailed t-test at $\alpha=0.05$:
Critical Value ($t_{critical}$): -1.6611
6. P-Value
One-tailed P-value: 0.0001
7. Decision
Reject $H_0$.

* P-value rule: $P\text{-value}=0.0001 < \alpha=0.05$.
* Critical value rule: $t=-3.9815 < t_{critical}=-1.6611$.

8. Conclusion
Connection of P-Value against $H_0$: Because the P-value (0.0001) is less than the significance level ($\alpha=0.05$), we reject $H_0$. This means that the probability of obtaining a sample mean of 1.3542 or lower purely by random chance when the true population mean is 2 is roughly 0.01%. Since this chance is well below our 5% threshold, there is statistically significant evidence to support $H_a$.
Verdict: Yes, we can conclude that the average number of dependents is significantly less than 2.

It is given that the population standard deviation (σ) for Sleep Duration is known and is equal to the sample standard deviation obtained from the dataset.


1.Is the average final GPA of the participants greater than 3.3 ?


In [ ]:
import numpy as np
from scipy.stats import norm


In [ ]:
# One-sample Z-test
z_stat = (final_gpa.mean() - 3.3) / (final_gpa.std(ddof=1) / np.sqrt(len(final_gpa)))

# One-tailed p-value (right-tailed)
p_value = 1 - norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a right-tailed test
critical_z_value = norm.ppf(1-alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(final_gpa),
        final_gpa.mean(),
        final_gpa.std(ddof=1),      # assumed population SD
        3.3,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,95.0000
1,Sample Mean,3.3254
2,Population Standard Deviation (σ),0.4374
3,Hypothesized Mean,3.3000
4,Critical value,1.6449
5,Z Statistic,0.5653
6,One-tailed p-value,0.2860


Hypothesis Test Evaluation
1. Null Hypothesis ($H_0$)$$H_0: \mu \le 3.3$$(The average final GPA of the population is less than or equal to 3.3.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu > 3.3$$(The average final GPA of the population is greater than 3.3.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-statistic with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary values:   Sample Size ($n$) = $95$   Sample Mean ($\bar{x}$) = $3.3254$   Population Standard Deviation ($\sigma$) = $0.4374$   Hypothesized Mean ($\mu_0$) = $3.3000$   $$Z = \frac{3.3254 - 3.3000}{\frac{0.4374}{\sqrt{95}}} = \frac{0.0254}{0.044876} \approx 0.5653$$   Test Statistic ($Z$): $0.5653$   
5. Critical ValueFor a right-tailed $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.2860$   
7. DecisionFail to reject $H_0$ (Do not reject the null hypothesis).
$P$-value rule: $P\text{-value} = 0.2860 > \alpha = 0.05$.   
Critical value rule: $Z = 0.5653 < Z_{\text{critical}} = 1.6449$ (the test statistic does not fall into the critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.2860$) is greater than the significance level ($\alpha = 0.05$), we fail to reject $H_0$. This means that the probability of obtaining a sample mean of 3.3254 or higher purely by random chance when the true population mean is 3.3 is roughly 28.6%. Since this chance is well above our 5% threshold, there is no statistically significant evidence to support $H_a$.

2.Is the average age of respondents greater than 28 years?

In [ ]:
# One-sample Z-test
z_stat = (age.mean() - 28) / (age.std(ddof=1) / np.sqrt(len(age)))

# One-tailed p-value (right-tailed)
p_value = 1 - norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a right-tailed test
critical_z_value = norm.ppf(1-alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(age),
        age.mean(),
        age.std(ddof=1),      # assumed population SD
        28,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,30.0000
2,Population Standard Deviation (σ),7.1185
3,Hypothesized Mean,28.0000
4,Critical value,1.6449
5,Z Statistic,2.7528
6,One-tailed p-value,0.0030


1. Null Hypothesis ($H_0$)$$H_0: \mu \le 28$$(The population mean age of respondents is less than or equal to 28 years.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu > 28$$(The population mean age of respondents is greater than 28 years.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $30.0000$   Population Standard Deviation ($\sigma$): $7.1185$   Hypothesized Mean ($\mu_0$): $28.0000$   $$Z = \frac{30.0000 - 28.0000}{\frac{7.1185}{\sqrt{96}}} = \frac{2.0000}{0.7265} \approx 2.7528$$   Test Statistic ($Z$): $2.7528$   
5. Critical ValueFor a one-tailed (right-tailed) $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.0030$   
7. DecisionReject $H_0$ (Reject the null hypothesis).
$P$-value rule: $P\text{-value} = 0.0030 < \alpha = 0.05$.   
Critical value rule: $Z = 2.7528 > Z_{\text{critical}} = 1.6449$ (the test statistic falls within the critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0030$) is less than the significance level ($\alpha = 0.05$), we reject $H_0$. A $P$-value of $0.0030$ indicates that if the true mean age were 28 years or less, there is only a 0.3% chance of observing a sample mean of 30.0000 years or higher purely by random chance. Since this probability is well below the 5% threshold, the result is statistically significant.

3.Does average number of certifications completed differ from 3?


In [ ]:
# One-sample Z-test
z_stat = (num_certs.mean() - 3) / (num_certs.std(ddof=1) / np.sqrt(len(num_certs)))

# Two-tailed p-value
p_value = 2*(1 - norm.cdf(abs(z_stat)))

alpha = 0.05

#Calculate critical values for a two-tailed test
lower_critical_z = norm.ppf(alpha / 2)
upper_critical_z = norm.ppf(1 - (alpha / 2))

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Lower Critical Z-value",
        "Upper Critical Z-value",
        "Z Statistic",
        "Two-tailed p-value"
    ],
    "Value": [
        len(num_certs),
        num_certs.mean(),
        num_certs.std(ddof=1),      # assumed population SD
        3,
        lower_critical_z,
        upper_critical_z,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,4.8438
2,Population Standard Deviation (σ),5.0353
3,Hypothesized Mean,3.0000
4,Lower Critical Z-value,-1.9600
5,Upper Critical Z-value,1.9600
6,Z Statistic,3.5877
7,Two-tailed p-value,0.0003


1. Null Hypothesis ($H_0$)$$H_0: \mu = 3$$(The average number of certifications completed is equal to 3.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu \neq 3$$(The average number of certifications completed differs from 3.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the two-tailed one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $4.8438$   Population Standard Deviation ($\sigma$): $5.0353$   Hypothesized Mean ($\mu_0$): $3.0000$   $$Z = \frac{4.8438 - 3.0000}{\frac{5.0353}{\sqrt{96}}} = \frac{1.8438}{0.5139} \approx 3.5877$$   Test Statistic ($Z$): $3.5877$   
5. Critical ValuesFor a two-tailed $Z$-test at $\alpha = 0.05$:Lower Critical $Z$-value: $-1.9600$   Upper Critical $Z$-value: $1.9600$   
6. $P$-ValueTwo-tailed $P$-value: $0.0003$   
7. DecisionReject $H_0$ (Reject the null hypothesis).$P$-value rule: $P\text{-value} = 0.0003 < \alpha = 0.05$.   
Critical value rule: $Z = 3.5877 > Z_{\text{upper}} = 1.9600$ (the test statistic falls into the upper critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0003$) is substantially less than the significance level ($\alpha = 0.05$), we reject $H_0$. A $P$-value of $0.0003$ indicates that if the true population mean number of certifications completed were equal to 3, there is only a 0.03% chance of observing a sample mean as far off as 4.8438 purely due to random variation. Since this probability is well below our 5% decision threshold, the observed difference is statistically significant.

4.Is the average years of experience greater than 5 years?

In [ ]:
# One-sample Z-test
z_stat = (experience.mean() - 5) / (experience.std(ddof=1) / np.sqrt(len(experience)))

# One-tailed p-value (right-tailed)
p_value = 1 - norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a right-tailed test
critical_z_value = norm.ppf(1-alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(experience),
        experience.mean(),
        experience.std(ddof=1),      # assumed population SD
        5,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,5.9479
2,Population Standard Deviation (σ),6.5351
3,Hypothesized Mean,5.0000
4,Critical value,1.6449
5,Z Statistic,1.4212
6,One-tailed p-value,0.0776


1. Null Hypothesis ($H_0$)$$H_0: \mu \le 5$$(The population mean years of experience is less than or equal to 5 years.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu > 5$$(The population mean years of experience is greater than 5 years.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $5.9479$   Population Standard Deviation ($\sigma$): $6.5351$   Hypothesized Mean ($\mu_0$): $5.0000$   $$Z = \frac{5.9479 - 5.0000}{\frac{6.5351}{\sqrt{96}}} = \frac{0.9479}{0.6670} \approx 1.4212$$   Test Statistic ($Z$): $1.4212$   
5. Critical ValueFor a one-tailed (right-tailed) $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.0776$   
7. DecisionFail to reject $H_0$ (Do not reject the null hypothesis).$P$-value rule: $P\text{-value} = 0.0776 > \alpha = 0.05$.   
Critical value rule: $Z = 1.4212 < Z_{\text{critical}} = 1.6449$ (the test statistic does not reach the critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0776$) is greater than the significance level ($\alpha = 0.05$), we fail to reject $H_0$. A $P$-value of $0.0776$ indicates that if the true population mean years of experience were 5 years, there is approximately a 7.76% probability of observing a sample mean of 5.9479 years or higher purely due to random sampling variation. Since this probability exceeds our 5% threshold, the result is not statistically significant.

5.Is the average number of relatives living abroad less than 5?

In [ ]:
# One-sample Z-test
z_stat = (relatives_abroad.mean() - 5) / (relatives_abroad.std(ddof=1) / np.sqrt(len(relatives_abroad)))

# One-tailed p-value (left-tailed)
p_value = norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a left-tailed test
critical_z_value = norm.ppf(alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(relatives_abroad),
        relatives_abroad.mean(),
        relatives_abroad.std(ddof=1),     # assumed population SD
        5,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,3.8854
2,Population Standard Deviation (σ),3.8821
3,Hypothesized Mean,5.0000
4,Critical value,-1.6449
5,Z Statistic,-2.8131
6,One-tailed p-value,0.0025


1. Null Hypothesis ($H_0$)$$H_0: \mu \ge 5$$(The average number of relatives living abroad is greater than or equal to 5.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu < 5$$(The average number of relatives living abroad is less than 5.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $3.8854$   Population Standard Deviation ($\sigma$): $3.8821$   Hypothesized Mean ($\mu_0$): $5.0000$   $$Z = \frac{3.8854 -
5.0000}{\frac{3.8821}{\sqrt{96}}} = \frac{-1.1146}{0.3962} \approx -2.8131$$   Test Statistic ($Z$): $-2.8131$   
5. Critical ValueFor a one-tailed (left-tailed) $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $-1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.0025$   
7. DecisionReject $H_0$ (Reject the null hypothesis).$P$-value rule: $P\text{-value} = 0.0025 < \alpha = 0.05$.   
Critical value rule: $Z = -2.8131 < Z_{\text{critical}} = -1.6449$ (the test statistic falls into the lower critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0025$) is less than the significance level ($\alpha = 0.05$), we reject $H_0$. A $P$-value of $0.0025$ indicates that if the true mean number of relatives living abroad were 5 or more, there is only a 0.25% chance of observing a sample mean of 3.8854 or lower purely due to random sampling variation. Since this probability is well below our 5% decision threshold, the result is statistically significant.

6.Is the average number of close friends in IT who have migrated greater than 5?

In [ ]:
# One-sample Z-test
z_stat = (friends_it_migrated.mean() - 5) / (friends_it_migrated.std(ddof=1) / np.sqrt(len(friends_it_migrated)))

# One-tailed p-value (right-tailed)
p_value = 1 - norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a right-tailed test
critical_z_value = norm.ppf(1-alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(friends_it_migrated),
        friends_it_migrated.mean(),
        friends_it_migrated.std(ddof=1),      # assumed population SD
        5,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,7.7812
2,Population Standard Deviation (σ),10.8086
3,Hypothesized Mean,5.0000
4,Critical value,1.6449
5,Z Statistic,2.5212
6,One-tailed p-value,0.0058


1. Null Hypothesis ($H_0$)$$H_0: \mu \le 5$$(The population mean number of close friends in IT who have migrated is less than or equal to 5.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu > 5$$(The population mean number of close friends in IT who have migrated is greater than 5.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $7.7812$   Population Standard Deviation ($\sigma$): $10.8086$   Hypothesized Mean ($\mu_0$): $5.0000$   $$Z = \frac{7.7812 - 5.0000}{\frac{10.8086}{\sqrt{96}}} = \frac{2.7812}{1.1031} \approx 2.5212$$   Test Statistic ($Z$): $2.5212$   
5. Critical ValueFor a one-tailed (right-tailed) $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.0058$   
7. DecisionReject $H_0$ (Reject the null hypothesis).$P$-value rule: $P\text{-value} = 0.0058 < \alpha = 0.05$.   
Critical value rule: $Z = 2.5212 > Z_{\text{critical}} = 1.6449$ (the test statistic falls into the right-tail rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0058$) is less than the significance level ($\alpha = 0.05$), we reject $H_0$. A $P$-value of $0.0058$ indicates that if the true population mean were 5 or less, there is only a 0.58% chance of observing a sample mean of 7.7812 or higher purely due to random sampling variation. Since this probability is well below our 5% threshold, the result is statistically significant.

7.Is the average number of dependents less than 2?

In [ ]:
# One-sample Z-test
z_stat = (num_of_dep.mean() - 2) / (num_of_dep.std(ddof=1) / np.sqrt(len(num_of_dep)))

# One-tailed p-value (left-tailed)
p_value = norm.cdf(z_stat)

alpha = 0.05

#Calculate critical value for a left-tailed test
critical_z_value = norm.ppf(alpha)

# Results
results = pd.DataFrame({
    "Statistic": [
        "Sample Size",
        "Sample Mean",
        "Population Standard Deviation (σ)",
        "Hypothesized Mean",
        "Critical value",
        "Z Statistic",
        "One-tailed p-value"
    ],
    "Value": [
        len(num_of_dep),
        num_of_dep.mean(),
        num_of_dep.std(ddof=1),     # assumed population SD
        2,
        critical_z_value,
        z_stat,
        p_value
    ]
})

display(results.round(4))

,Statistic,Value
0,Sample Size,96.0000
1,Sample Mean,1.3542
2,Population Standard Deviation (σ),1.5893
3,Hypothesized Mean,2.0000
4,Critical value,-1.6449
5,Z Statistic,-3.9815
6,One-tailed p-value,0.0000


1. Null Hypothesis ($H_0$)$$H_0: \mu \ge 2$$(The average number of dependents is greater than or equal to 2.)   
2. Alternative Hypothesis ($H_a$)$$H_a: \mu < 2$$(The average number of dependents is less than 2.)   
3. Significance Level ($\alpha$)$$\alpha = 0.05$$
4. Test StatisticUsing the formula for the one-sample $Z$-test with population standard deviation ($\sigma$):$$Z = \frac{\bar{x} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$Substituting the provided summary statistics:   Sample Size ($n$): $96$   Sample Mean ($\bar{x}$): $1.3542$   Population Standard Deviation ($\sigma$): $1.5893$   Hypothesized Mean ($\mu_0$): $2.0000$   $$Z = \frac{1.3542 - 2.0000}{\frac{1.5893}{\sqrt{96}}} = \frac{-0.6458}{0.1622} \approx -3.9815$$   Test Statistic ($Z$): $-3.9815$   
5. Critical ValueFor a one-tailed (left-tailed) $Z$-test at $\alpha = 0.05$:Critical Value ($Z_{\text{critical}}$): $-1.6449$   
6. $P$-ValueOne-tailed $P$-value: $0.0000$ ($P < 0.0001$)   
7. DecisionReject $H_0$ (Reject the null hypothesis).$P$-value rule: $P\text{-value} = 0.0000 < \alpha = 0.05$.   
Critical value rule: $Z = -3.9815 < Z_{\text{critical}} = -1.6449$ (the test statistic falls far into the lower critical rejection region).   
8. ConclusionConnection of $P$-Value against $H_0$: Because the $P$-value ($0.0000$) is less than the significance level ($\alpha = 0.05$), we reject $H_0$. A $P$-value of $0.0000$ indicates that if the true population mean number of dependents were 2 or more, there is virtually a 0% chance ($< 0.01\%$) of obtaining a sample mean as low as 1.3542 purely due to random sampling variation. Since this probability is far below our 5% threshold, the result is highly statistically significant.

### Critical Z-value Calculation for One-tailed Tests

In [ ]:
import numpy as np
from scipy.stats import norm

alpha = 0.05

# For a right-tailed test (e.g., to see if a mean is GREATER than a hypothesized value)
# The critical value is at 1 - alpha
right_tailed_critical_z = norm.ppf(1 - alpha)

# For a left-tailed test (e.g., to see if a mean is LESS than a hypothesized value)
# The critical value is at alpha
left_tailed_critical_z = norm.ppf(alpha)

print(f"For alpha = {alpha} (one-tailed test):")
print(f"Right-tailed critical Z-value (for H1: μ > μ0): {right_tailed_critical_z:.4f}")
print(f"Left-tailed critical Z-value (for H1: μ < μ0): {left_tailed_critical_z:.4f}")

For alpha = 0.05 (one-tailed test):
Right-tailed critical Z-value (for H1: μ > μ0): 1.6449
Left-tailed critical Z-value (for H1: μ < μ0): -1.6449
